calico/felix/dispatch.py

```diff
@@ -47,6 +47,7 @@ def __init__(self, config, ip_version, iptables_updater):
         self.ifaces = set()
         self.programmed_leaf_chains = set()
         self._dirty = False
+        self._datamodel_in_sync = False
```

Comment:
I found the name of this variable confusing. It really means "is there a snapshot in progress", but I read it as "we have no changes to make". Worth renaming it, I think.